tilavarauspalvelu/urls.py

```diff
@@ -1,20 +1,25 @@
 from django.conf import settings
 from django.conf.urls.static import static
 from django.contrib import admin
-from django.urls import include, path
+from django.urls import include, path, reverse
 from django.views.decorators.csrf import csrf_exempt
 from graphene_django_extensions import FileUploadGraphQLView
 
 from api.legacy_rest_api.urls import legacy_outer
 from api.webhooks.urls import webhook_router
 
-# Mock the autocomplete view to be able to include `extra_context` for other pages
-real_autocomplete_view = admin.site.autocomplete_view
-admin.site.autocomplete_view = lambda request, extra_context: real_autocomplete_view(request)
-
-# Mock the catch-all view to be able to include `extra_context` for other pages
-real_catch_all_view = admin.site.catch_all_view
-admin.site.catch_all_view = lambda request, url, extra_context: real_catch_all_view(request, url)
+# Mock the `each_context` method to add some custom context variables.
+original_each_context = admin.site.each_context
+admin.site.each_context = lambda request: original_each_context(request) | {
+    "version": settings.APP_VERSION,
+    # The helauth variables need to be added, since we subclass `helusers.tunnistamo_oidc.TunnistamoOIDCAuth`
+    # with `tilavarauspalvelu.auth.ProxyTunnistamoOIDCAuthBackend` for optimizing request user fetching.
+    # `helusers.admin_site.AdminSite.each_context` refers to the original backend by
+    # string reference, so subclasses won't have the login/logout urls added.
+    "helsinki_provider_installed": True,
+    "helsinki_login_url": reverse("helusers:auth_login"),
+    "helsinki_logout_url": reverse("helusers:auth_logout"),
+}
```

Comment:
There is a bunch of additional setup that needs to be made if we make it a subclass, since we would need to make our admin site the default admin site. I think this is clear enough to not warrant the additional setup.